ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -94,16 +99,43 @@ let networks path (list: string -> string list) =
         let after_prefix = String.sub str prefix_len size in
         after_prefix
       in
+      (* get vif id of the eth which own these mac addresses *)
+      let rec find_vif_id mac_list vifs =
+        match vifs with
+        | [] -> None
+        | vif_id :: rest ->
+          match lookup (Printf.sprintf "device/vif/%s/mac" vif_id) with
+          | None -> find_vif_id mac_list rest
+          | Some mac_of_vif -> 
+            if List.mem mac_of_vif mac_list then begin
+              Some vif_id
+            end else
+              find_vif_id mac_list rest
+      in
+      (* get all mac addresses of eth *)
+      let get_mac_list eth =
+        let key_prefix = Printf.sprintf "attr/%s/mac" eth in
+        list key_prefix
+        |> Listext.List.filter_map (fun mac_id -> lookup (Printf.sprintf "%s/%s" key_prefix mac_id))
```

Comment:
`Listext` was opened above so there would be no need to use it here. However, I think it adds clarity to qualify values explicitly as it is done here and to not open `Listext`.